### PR TITLE
Run tests in parallel

### PR DESCRIPTION
**Pull request by attiasas**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [x] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----

# TODO:
1. Update actions and use cache
2. Re structure test data to better places (also move npm folder... move artifactory configs to others...)
image
3. Try move integration tests to its own new dir `integrationtests` in `tests`

## Running Tests in Parallel, Splitting Based on Types

* This approach reduces run time from approximately 30 minutes to about 15 minutes.
* Adding `-race` missing flag for tests
* It enables running partial tests locally by types or executing all of them using simple flags.

### Confirmation: All Tests Are Passing (Note: This will only pass after the merge)

image

### CODE REVIEW

File: tests/utils/test_utils.go [hunk 1]
```diff
@@ -16,9 +16,42 @@ import (
...
-		t.Skip("Skipping Security test. To run Security test add the '-test.security=true' option.")
+func InitUnitTest(t *testing.T) {
+	if !*configTests.TestUnit {
+		t.Skip("Skipping unit tests.")
+	}
+}
+
+func InitArtifactoryTest(t *testing.T) {
+	if !*configTests.TestArtifactory {
+		t.Skip("Skipping Artifactory test. To run Artifactory test add the '-test.artifactory=true' option.")
```
Comment (eranturgeman) [inline tests/utils/test_utils.go:27]:
```suggestion
		t.Skip("Skipping Artifactory tests. To run Artifactory tests, add the '-test.artifactory=true' option.")
```

File: tests/utils/test_utils.go [hunk 2]
```diff
@@ -16,9 +16,42 @@ import (
...
+	}
+}
+
+func InitXrayTest(t *testing.T, minVersion string) {
+	if !*configTests.TestXray {
+		t.Skip("Skipping Xray test. To run Xray test add the '-test.xray=true' option.")
```
Comment (eranturgeman) [inline tests/utils/test_utils.go:33]:
```suggestion
		t.Skip("Skipping Xray tests. To run Xray tests, add the '-test.xray=true' option.")
```

File: tests/utils/test_utils.go [hunk 3]
```diff
@@ -16,9 +16,42 @@ import (
...
+	}
+	ValidateXrayVersion(t, minVersion)
+}
+
+func InitAuditTest(t *testing.T, minVersion string) {
+	if !*configTests.TestAudit {
+		t.Skip("Skipping audit test. To run Audit test add the '-test.audit=true' option.")
```
Comment (eranturgeman) [inline tests/utils/test_utils.go:40]:
```suggestion
		t.Skip("Skipping audit tests. To run Audit tests, add the '-test.audit=true' option.")
```

File: tests/utils/test_utils.go [hunk 4]
```diff
@@ -16,9 +16,42 @@ import (
...
+	}
+	ValidateXrayVersion(t, minVersion)
+}
+
+func InitScanTest(t *testing.T, minVersion string) {
+	if !*configTests.TestScan {
+		t.Skip("Skipping scan test. To run Scan test add the '-test.scan=true' option.")
```
Comment (eranturgeman) [inline tests/utils/test_utils.go:47]:
```suggestion
		t.Skip("Skipping scan tests. To run Scan tests, add the '-test.scan=true' option.")
```

File: tests/utils/test_utils.go [hunk 5]
```diff
@@ -16,9 +16,42 @@ import (
...
+	}
+	ValidateXrayVersion(t, minVersion)
+}
+
+func InitDockerScanTest(t *testing.T, minVersion string) {
+	if !*configTests.TestDockerScan || !*configTests.TestScan {
+		t.Skip("Skipping Docker scan test. To run Xray Docker test add the '-test.dockerScan=true' and '-test.scan=true' options.")
```
Comment (eranturgeman) [inline tests/utils/test_utils.go:54]:
```suggestion
		t.Skip("Skipping Docker scan tests. To run Xray Docker tests, add the '-test.dockerScan=true' and '-test.scan=true' options.")
```

File: tests/config.go [hunk 6]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
 	"github.com/jfrog/jfrog-client-go/auth"
 	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	clientLog "github.com/jfrog/jfrog-client-go/utils/log"
```
Comment (eranturgeman) [inline tests/config.go:11]: del empty line

File: tests/config.go [hunk 7]
```diff
@@ -41,13 +47,16 @@ var (
```
Comment (eranturgeman) [inline tests/config.go:47]: del empty line

File: tests/config.go [hunk 8]
```diff
@@ -41,13 +47,16 @@ var (
 
 	ContainerRegistry *string
```
Comment (eranturgeman) [inline tests/config.go:49]: del empty line

File: tests/config.go [hunk 9]
```diff
@@ -59,8 +68,19 @@ func init() {
```
Comment (eranturgeman) [inline tests/config.go:68]: del empty line

**Review by eranturgeman — COMMENTED** [review]
Just make sure to organize dependencies in the files you worked in

**Comment by attiasas** [comment]
Implemented at:
* https://github.com/jfrog/jfrog-cli-security/pull/162

**Pull request closed without merging**